### PR TITLE
Add new blog post about Kudos mapping

### PR DESCRIPTION
**Pull request by sherrardTr4129**

### CODE REVIEW

**Comment by liam-middlebrook** [comment]
@jefftaglic here's a link that explains how to do in-line reviews:
https://help.github.com/articles/reviewing-proposed-changes-in-a-pull-request/

**Comment by sherrardTr4129** [comment]
Jeff, I fixed those spelling/formatting issues per your review. 

I use that slogan on my personal website and professional profiles. I appreciate your input, but I think I'm going to keep that as it is for now.

File: _posts/2018-04-01-kudos-mapping-and-slam.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
+layout: post
+title: Kudos Mapping and SLAM
+date: 2018-04-01
+categories:
```
Comment (csssuf) [inline _posts/2018-04-01-kudos-mapping-and-slam.md]: Add `project`, I think?

File: _posts/2018-04-01-kudos-mapping-and-slam.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    - Robotics
+    - SLAM
+    - ROS
+    - Mapping
+description: Added LiDAR mapping functionality to Computer Science House's floor robot, Kudos.  
+image: https://github.com/ComputerScienceHouse/Kudos/blob/master/kudos_csh/images/CSH_Map_2.png?raw=true
+image-sm: http://www.trevorsherrard.com/images/CSH_Map_2.png
+author: Trevor Sherrard
+author-image: http://www.trevorsherrard.com/images/profile_pics.jpg
+author-bio: Electrical Engineer, roboticist, computer vision enthusiast. I have a strong passion seeing my software come alive in my projects.
```
Comment (csssuf) [inline _posts/2018-04-01-kudos-mapping-and-slam.md]: I know this has been touched on elsewhere, but I think "I have a strong passion *for* seeing my software come alive in my projects" would work better across the board.

File: _posts/2018-04-01-kudos-mapping-and-slam.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+author-email: tws4129@rit.edu
+author-social:
+    github: https://github.com/sherrardTr4129
+    linkedin: https://www.linkedin.com/in/trevor-sherrard
+---
+
+## Background
+Computer Science House purchased a robot chasis from one of our members in the Spring semester of 2017. I had worked on previous versions of the CSH H.E.R.O.I.N project and thought I would be able to make significant strides given we now had a stable robot chasis. So, using robot operating system constructs and several new pieces of hardware/sensors, I added mapping functionality to Kudos.
```
Comment (csssuf) [inline _posts/2018-04-01-kudos-mapping-and-slam.md]: `s/chasis/chassis/`

File: _posts/2018-04-01-kudos-mapping-and-slam.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+## Details
+First, I needed to install a LiDAR sensor on the robot. Standard off the self LiDAR solutions, such as the 16-channel velodyne puck, start at upwards of $5000. Being a student organization, we needed a more financially sensible solution. I decided on using a replacement LiDAR sensor from the NeatoXV robot vaccum cleaner. Once the serial communication between the LiDAR controller board was setup with the Raspberry Pi, ROS development began. The LiDAR scans were "off-boarded" the robot via the rostopic /laserscan via the network to a VM running on CSH proxmox. On this VM, a program subscribed to this data and performed an algorithm to simultaneously localize the robot within a map and build the map as the robot moved through its enviornment. A navigation goal could be set for the robot to navigate to, and a path to follow to obtain this goal was computed. From this path, individual robot movements were calculated and streamed to the robot via Computer Science House's WiFi network. This robot movement was actuated upon by the robot through its move base controller. See the figure below for a drawing of the control topology for Kudos.
```
Comment (csssuf) [inline _posts/2018-04-01-kudos-mapping-and-slam.md]: `s/off the self/off-the-shelf/`
`s/velodyne/Velodyne/`
`s/vaccum/vacuum/`
`s/"off-boarded" the robot/"off-boarded" from the robot/`
Using "via" twice in a row like this is confusing, I'd try to reword this.
`s/proxmox/Proxmox/`

File: _posts/2018-04-01-kudos-mapping-and-slam.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+![Kudos Control Topology](http://www.trevorsherrard.com/images/KudosCommProtocol.png)
+
+## Future Iterations
+For anyone thinking about working on this project in future student generations, make sure to develop/use a SLAM algorithm that takes it odometry data from encoders. Currently the software setups relies on a fairly consistent LiDAR scan update rate and a large difference between LiDAR scan frames. Unfortunately, most walls are very flat and not very feature-rich. Even more unfortunate, most of CSH is comprised of very flat walls, so Kudos had some difficulty navigating when traveling through long, straight hallways. Kudos has gearboxes that allow for easy install of AndyMark standard encoders.
```
Comment (csssuf) [inline _posts/2018-04-01-kudos-mapping-and-slam.md]: `s/takes it odometry/takes its odometry/`
`s/software setups/software setup/`

**Comment by sherrardTr4129** [comment]
Should be good now. Let me know what you think.

**Comment by csssuf** [comment]
Make sure you update the date in both the filename and the front matter before merging.

**Comment by sherrardTr4129** [comment]
Date is changed. Should be all set.

**Pull request merged**